mitiq/shadows/classical_postprocessing.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+"""Classical post-processing process of classical shadows."""
+
+from typing import Tuple, List, Any
+
+import cirq
+import numpy as np
+from numpy.typing import NDArray
+
+# local unitary that applied to the qubits
+phase_z = cirq.S._unitary_().conj()
+hadamard = cirq.H._unitary_()
+identity = cirq.I._unitary_()
+PAULI_MAP = {"X": hadamard, "Y": hadamard @ phase_z, "Z": identity}
+
+
+def snapshot_state(b_list: List[float], u_list: List[str]) -> NDArray[Any]:
+    r"""
+    Implement a single snapshot state reconstruction.
+
+    Args:
+        b_list: The list of classical outcomes for the snapshot. Here,
+            b = 1 corresponds to :math:`|0\rangle`, and
+            b = -1 corresponds to :math:`|1\rangle`.
+
+        u_list: Array of ("X", "Y", "Z") for the applied
+            Pauli measurement on each qubit.
+
+    Returns:
+        Reconstructed snapshot in terms of nparray.
+    """
+
+    # z-basis measurement outcomes
+    b_zero = np.array([[1.0 + 0.0j, 0.0 + 0.0j]])
+    zero_state = b_zero.T @ b_zero
+
+    b_one = np.array([[0.0 + 0.0j, 1.0 + 0.0j]])
+    one_state = b_one.T @ b_one
+
+    # reconstructing a single snapshot state by applying Eq. (S44)
+    rho_snapshot = np.array([1.0 + 0.0j])
+
+    for b, u in zip(b_list, u_list):
+        state = zero_state if b == 1 else one_state
+        U = PAULI_MAP[u]
+        # act $$Ad_{U^\dagger}$$ on the computational basis states
+        local_rho = 3.0 * (U.conj().T @ state @ U) - identity
+        rho_snapshot = np.kron(rho_snapshot, local_rho)
+
+    return rho_snapshot
+
+
+def shadow_state_reconstruction(
+    measurement_outcomes: Tuple[NDArray[Any], NDArray[np.str0]]
+) -> NDArray[Any]:
+    """Reconstruct a state approximation as an average over all snapshots.
+
+    Args:
+        measurement_outcomes: A shadow tuple obtained
+        from `z_basis_measurement`.
+
+    Returns:
+        Numpy array with the reconstructed quantum state.
+    """
+    num_snapshots, num_qubits = measurement_outcomes[0].shape
+
+    # classical values
+    b_lists, u_lists = measurement_outcomes
+
+    # Averaging over snapshot states.
+    shadow_rho = np.zeros((2**num_qubits, 2**num_qubits), dtype=complex)
+    for i in range(num_snapshots):
+        shadow_rho += snapshot_state(b_lists[i], u_lists[i])
```

Comment:
Not sure if `bits` and `op` are good variable names, but WDYT of this?
```suggestion
    shadow_rho = sum(
        snapshot_state(bits, op) for bits, op in zip(b_lists, u_lists)
    )
```

Reply:
Thanks! I just changed it to 
```python
    shadow_rho = np.sum(
        [
            classical_snapshot(b_list, u_list)
            for b_list, u_list in zip(b_lists, u_lists)
        ],
        axis=0,
    )
```
in a new commit

Reply:
Hmm, doesn't adding `axis=0` change the effect of this sum, from the original `for` loop?

Reply:
I'll get errors as your suggested code. I think it's because if you delete axis=0 from np.sum(), numpy will sum all the elements in the array, resulting in a single scalar value, rather than an array.